### PR TITLE
refactor(robot-server): Remove Pydantic `.construct()`s in tests

### PR DESCRIPTION
**Pull request by SyntaxColoring**
This PR replaces, in test files, Pydantic `Model.construct(...)` with `Model(...)`. Before this PR, we had an inconsistent mixture of both.

This was decided previously on the CPX squad. Our rationale was roughly:

* The performance benefits of `.construct()` are not worthwhile in tests
* The runtime validation in tests is a nice final defense in case our production code does something that subverts static type-checking, like `.update({...})`, and it gets it wrong somehow.

`Model.construct(...)` will remain in the production code, for performance reasons.

# Review requests

I used a global search and replace here. Skim over the code and make sure there's nothing obviously wrong.

# Risk assessment

Very low. Changes are only to test files

### CODE REVIEW

**Comment by SyntaxColoring** [comment]
Uh okay it looks like this broke a lot:

* `decoy.matchers`, which sort of makes sense.
* New errors about missing fields? Does the Pydantic plugin for Mypy not warn us if fields are missing in `.construct()`? If so, that's pretty horrifying.
* Other new errors? Why is `tests/runs/router/test_commands_router.py::test_get_run_commands` failing with what looks like an actual data mismatch?

File: api/tests/opentrons/protocol_engine/state/test_module_store.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_initial_state() -> None:
 def test_load_module(tempdeck_v2_def: ModuleDefinition) -> None:
     """It should handle a successful LoadModule command."""
     action = actions.UpdateCommandAction(
-        command=commands.LoadModule.construct(  # type: ignore[call-arg]
+        command=commands.LoadModule(  # type: ignore[call-arg]
```
Comment (mcous) [inline api/tests/opentrons/protocol_engine/state/test_module_store.py]: This was construct due to a bunch of missing required parameters, which is also what the `# type: ignore[call-arg]` is there for

File: api/tests/opentrons/protocol_runner/smoke_tests/test_json_file_reader.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ async def test_reads_file(
     subject = JsonFileReader()
     result = subject.read(protocol_source)
 
-    assert result == json_protocol.Model.construct(
+    assert result == json_protocol.Model(
```
Comment (mcous) [inline api/tests/opentrons/protocol_runner/smoke_tests/test_json_file_reader.py]: This uses a `matchers.IsA(dict)` down on line 43

**Review by mcous — CHANGES_REQUESTED** [review]
Only commented on a couple demonstration lines, but everything that either...

- Uses a `matchers` object, or
- Omits required parameters that aren't used by the code under test
    - These lines should have `type-ignore[call-args]` comments 
    - Aside: this for sure a questionable practice, be we can leave addressing that outside the scope of this PR

...should be using `construct`.

I think that means from this PR, the only changes we want to be making are in `robot-server`. That makes sense to me, I bet they were caught up in a find-and-replace-all or something

**Review by mcous — APPROVED** [review]
Reverted the `construct` removals that should not have happened so we can close this PR out

**Pull request merged**